File: app/schemas/email.py
"""Pydantic models for API responses."""

from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel


class EmailSchema(BaseModel):
    id: int
    provider_id: str
    thread_id: str | None
    sender: str
    recipients: str
    cc: str | None
    subject: str
    snippet: str | None
    body: str
    received_at: datetime
    processing_status: str
    lead_flag: bool
    category: str | None
    priority: str | None
    extracted_entities: Optional[dict[str, Any]]
    suggested_reply: str | None
    reply_generated_at: datetime | None

    class Config:
        from_attributes = True


class EmailListResponse(BaseModel):
    emails: list[EmailSchema]


class EmailSyncResponse(BaseModel):
    synced: int


class ReTriageResponse(BaseModel):
    email: EmailSchema

File: frontend/src/types/email.ts
export interface Email {
    id: number;
    provider_id: string;
    thread_id?: string | null;
    sender: string;
    recipients: string;
    cc?: string | null;
    subject: string;
    snippet?: string | null;
    body: string;
    received_at: string; // ISO date string
    processing_status: string;
    lead_flag: boolean;
    category?: string | null;
    priority?: string | null;
    extracted_entities?: Record<string, any> | null;
    suggested_reply?: string | null;
    reply_generated_at?: string | null;
}

export interface EmailListResponse {
    emails: Email[];
}

export interface SyncResponse {
    success: boolean;
    synced: number;
    classified?: number;
    replies_generated?: number;
    error?: string;
}

export interface ResetResponse {
    success: boolean;
    deleted: number;
    error?: string;
}

